.circleci/config.yml

```diff
@@ -22,60 +20,174 @@ jobs:
-          command: |
-            nyc check-coverage --statements=100 --functions=100 --branches=100 --lines=100
-            nyc check-coverage --temp-directory ./jscov/client/ --statements=100 --functions=100 --branches=100 --lines=100
+          command: nyc check-coverage --statements=100 --functions=100 --branches=100 --lines=100
+      - store_artifacts:
+          path: jscov
+          destination: js-reports
+      - store_test_results:
+          path: jscov
+
+  test-client:
+    working_directory: ~/herman
+    docker:
+      - image: circleci/node:8-browsers
+    steps:
+      - checkout
+      - restore_cache:
+          key: herman-node-modules-{{ checksum "yarn.lock" }}
+      - run:
+          name: Set $PATH
+          command: echo 'export PATH=~/herman/node_modules/.bin:$PATH' >> $BASH_ENV
+      - run:
+          name: Run Client JS Tests & Check Coverage
+          command: gulp clienttest
+      # Do not report multiple times to coveralls: https://github.com/nickmerwin/node-coveralls/issues/176
+      # - run:
+      #     name: Report Test Coverage
+      #     command: nyc report --reporter=text-lcov --temp-directory ./jscov/client/ | coveralls
+      - run:
+          name: Check Test Coverage
+          command: nyc check-coverage --temp-directory ./jscov/client/ --statements=100 --functions=100 --branches=100 --lines=100
+      - store_artifacts:
+          path: jscov
+          destination: js-reports
+      - store_test_results:
+          path: jscov
+
+  test-src-node-6:
+    working_directory: ~/herman
+    docker:
+      - image: node:6
+    steps:
+      - checkout
+      - restore_cache:
+          key: herman-node6-modules-{{ checksum "yarn.lock" }}
+      - run:
+          name: Set $PATH
+          command: echo 'export PATH=~/herman/node_modules/.bin:$PATH' >> $BASH_ENV
       - run:
           name: Run JS Tests [node 6]
-          command: |
-            export NVM_DIR="$HOME/.nvm"
-            [ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"
-            nvm install 6.2.2
-            yarn install --force
-            gulp jstest
-            nyc check-coverage --statements=100 --functions=100 --branches=100 --lines=100
-      - deploy:
-          name: Update yarn.lock [greenkeeper]
-          command: |
-            if [[ "${CIRCLE_BRANCH}" =~ greenkeeper\/.* ]]; then
-              yarn global add greenkeeper-lockfile@1 --prefix ~/.yarn
-              ~/.yarn/bin/greenkeeper-lockfile-update
-              ~/.yarn/bin/greenkeeper-lockfile-upload
-            fi
+          command: gulp jstest
+      # Do not report multiple times to coveralls: https://github.com/nickmerwin/node-coveralls/issues/176
+      # - run:
+      #     name: Report Test Coverage [node 6]
+      #     command: nyc report --reporter=text-lcov | coveralls
+      - run:
+          name: Check Test Coverage [node 6]
+          command: nyc check-coverage --statements=100 --functions=100 --branches=100 --lines=100
```

Comment:
As a philsophical point, I feel that anything we can push from `.circleci/config` into in-repo configs for the tools in question, we should. That way, when we run the tools locally, we get the same options set. Of course, Gulp handles a lot of that for us at the moment, which is why I say this is philosophical, but I think worth saying.

Reply:
Yeah, that's a good point. At this point we're only enforcing coverage on CI, and I don't really want to do that locally in any case. But it wouldn't hurt to pull the settings into `gulpfile.js` or another config, so that CI doesn't contain the canonical settings.